cube/src/main/java/zone/cogni/asquare/cube/index/ElasticsearchIndexingService.java

```diff
@@ -0,0 +1,604 @@
+
+    if (indexNames.isEmpty()) {
+      log.error("{} no valid indexes configured", indent);
+    }
+
+    if (!indexNames.isEmpty())
+      log.info("{} valid indexes:   {}", indent, String.join(", ", indexNames));
+    if (!getInvalidIndexNames().isEmpty())
+      log.info("{} invalid indexes: {}", indent, String.join(", ", getInvalidIndexNames()));
+
+    indexesMetadata.forEach(this::validateIndex);
+  }
+
+  private void validateIndex(IndexMetadata indexMetadata) {
+    // valid or not
+    if (indexMetadata.isValid())
+      log.info("  '{}' index is valid", indexMetadata.getName());
+    else
+      log.error("  '{}' index NOT is valid", indexMetadata.getName());
+
+    // settings
+    if (!indexMetadata.isValidSettingsResource())
+      log.error("{}   elastic-settings.json is missing", indent);
+
+    // collections
+    List<String> collectionNames = getValidCollectionNames(indexMetadata);
+    log.info("{}   collection count: {}", indent, collectionNames.size());
+
+    if (collectionNames.isEmpty()) {
+      log.error("{}   no valid collections configured", indent);
+    }
+
+    if (!collectionNames.isEmpty())
+      log.info("{}   valid collections:   {}", indent, String.join(", ", collectionNames));
+    if (!getInvalidCollectionNames(indexMetadata).isEmpty())
+      log.info("{}   invalid collections: {}", indent, String.join(", ", getInvalidCollectionNames(indexMetadata)));
+
+    // deeper check into collections
+    indexMetadata.getCollections().forEach(this::validateCollection);
+  }
+
+  private void validateCollection(CollectionMetadata collectionMetadata) {
+    // valid or not
+    if (collectionMetadata.isValid())
+      log.info("    '{}' collection is valid", collectionMetadata.getName());
+    else
+      log.error("    '{}' collection NOT is valid", collectionMetadata.getName());
+
+    // select
+    if (collectionMetadata.getSelectQueryResources().isEmpty())
+      log.error("{}     select-* queries are missing", indent);
+
+    // construct
+    if (collectionMetadata.getConstructQueryResources().isEmpty())
+      log.error("{}     construct-* queries are missing", indent);
+
+    // facet
+    if (collectionMetadata.getFacetQueryResources().isEmpty())
+      log.warn("{}     facets/* queries are missing", indent);
+  }
+
+  /**
+   * @return <code>true</code> if indexing is active
+   */
+  public boolean isIndexRunning() {
+    return indexMonitoredPool.isActive();
+  }
+
+  /**
+   * @return list of indexes managed by service for current <code>elasticStore</code>
+   */
+  @Nonnull
+  public List<String> getIndexNames() {
+    return indexesMetadata.stream()
+                          .filter(IndexMetadata::isValid)
+                          .map(IndexMetadata::getName)
+                          .collect(Collectors.toList());
+  }
+
+  private List<String> getInvalidIndexNames() {
+    return indexesMetadata.stream()
+                          .filter(indexMetadata -> !indexMetadata.isValid())
```

Comment:
Fun one (might be added in later java version)
add method somewhere:
```  
public static <T> Predicate<T> not(Predicate<T> t) {
    return t.negate();
}
  ```
  
then do this:
`.filter(not(IndexMetadata::isValid))`

Reply:
weird you have to do it yourself. in jdk11 there is a method btw.